variables.tf

```
@@ -818,10 +820,28 @@ variable "runner_worker_docker_autoscaler_asg" {
     spot_allocation_strategy                 = optional(string, "lowest-price")
     spot_instance_pools                      = optional(number, 2)
     subnet_ids                               = optional(list(string), [])
-    types                                    = optional(list(string), ["m5.large"])
+    default_instance_type                    = optional(string, "m5.large")
+    types                                    = optional(list(string), [])
     upgrade_strategy                         = optional(string, "rolling")
+    instance_requirements = optional(list(object({
+      allowed_instance_types = optional(list(string), [])
+      cpu_manufacturers      = optional(list(string), [])
+      instance_generations   = optional(list(string), [])
+      burstable_performance  = optional(string)
+      memory_mib = optional(object({
+        min = optional(number, null)
+      max = optional(number, null) }), {})
+      vcpu_count = optional(object({
+        min = optional(number, null)
+      max = optional(number, null) }), {})
+    })), [])
   })
   default = {}
+
+  validation {
+    condition     = !(length(var.runner_worker_docker_autoscaler_asg.types) > 0 && length(var.runner_worker_docker_autoscaler_asg.instance_requirements) > 0)
```

suggestion: remove the negation to reduce complexity.

```suggestion
    condition     = length(var.runner_worker_docker_autoscaler_asg.types) = 0 ||length(var.runner_worker_docker_autoscaler_asg.instance_requirements) = 0
```